cmd/sync/sync.go

```diff
@@ -3,64 +3,232 @@
+	Type               string      `json:"type"`
+	User               string      `json:"user"`
+	Filepath           string      `json:"filepath"`
+	AccessionID        string      `json:"accession_id"`
+	DecryptedChecksums []checksums `json:"decrypted_checksums"`
+}
+
+// Checksums is struct for the checksum type and value
+type checksums struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+// Completed is struct holding the full message data
+type completed struct {
+	User               string      `json:"user"`
+	Filepath           string      `json:"filepath"`
+	AccessionID        string      `json:"accession_id"`
+	DecryptedChecksums []checksums `json:"decrypted_checksums"`
 }
 
 func main() {
-	conf, err := config.NewConfig("ingest")
+	conf, err := config.NewConfig("sync")
 	if err != nil {
 		log.Fatal(err)
 	}
-
-	archive, err := storage.NewBackend(conf.Archive)
+	mq, err := broker.NewMQ(conf.Broker)
 	if err != nil {
 		log.Fatal(err)
-
 	}
+	db, err := database.NewDB(conf.Database)
+	if err != nil {
+		log.Fatal(err)
+	}
+	backup, err := storage.NewBackend(conf.Backup)
+	if err != nil {
+		log.Fatal(err)
 
+	}
 	inbox, err := storage.NewBackend(conf.Inbox)
 	if err != nil {
 		log.Fatal(err)
 
 	}
 
+	defer mq.Channel.Close()
+	defer mq.Connection.Close()
+	defer db.Close()
+
+	go func() {
+		connError := mq.ConnectionWatcher()
+		log.Error(connError)
+		os.Exit(1)
+	}()
+
 	forever := make(chan bool)
 
-	var message trigger
-	message.Filepath = "Web.pdf"
+	log.Info("starting sync service")
+	var message sync
+
+	go func() {
+		messages, err := mq.GetMessages(conf.Broker.Queue)
+		if err != nil {
+			log.Fatal(err)
+		}
+		for delivered := range messages {
+			log.Debugf("received a message: %s", delivered.Body)
+			res, err := validateJSON(conf.SchemasPath, delivered.Body)
+			if err != nil {
+				log.Errorf("josn error: %v", err)
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorln("failed to Nack message, reason: ", e)
+				}
+
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendJSONError(&delivered, err.Error(), conf.Broker); e != nil {
+					log.Error("failed to publish message, reason: ", err)
+				}
+				// Restart on new message
+				continue
+			}
+			if !res.Valid() {
+				log.Errorf("result.error: %v", res.Errors())
+				log.Error("Validation failed")
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorln("failed to Nack message, reason: ", e)
+				}
+
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendJSONError(&delivered, err.Error(), conf.Broker); e != nil {
+					log.Error("failed to publish message, reason: ", res.Errors())
+				}
+				// Restart on new message
+				continue
+			}
```

Comment:
The logging additions we merged recently also includes moving roughly this to broker, so this can be shortened to a single call.